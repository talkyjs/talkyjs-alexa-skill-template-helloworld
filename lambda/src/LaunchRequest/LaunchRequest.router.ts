import { Router } from "@talkyjs/core";

import { LaunchRequestScript } from './LaunchRequest.speech'


export const LaunchRequestRouter: Router = {
    requestType: "LaunchRequest",
    
    handler: async (handlerInput) => {
        
        
        const script = new LaunchRequestScript(handlerInput, {
            dummyProps: "It's a nice development."
        })
        return script
            .createResponseBuilder()
            .getResponse();
        
        
    }
}

export default LaunchRequestRouter