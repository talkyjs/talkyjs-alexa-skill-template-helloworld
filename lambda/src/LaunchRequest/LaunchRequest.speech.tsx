import React from 'react';
import { SpeechScriptJSXWithOption } from '@talkyjs/ssml'

export class LaunchRequestScript extends SpeechScriptJSXWithOption<{
    dummyProps: string;
}> {
    speech() {
        return (
            <speak>
                <p>
                    <amazon-emotion name="excited" intensity="high">Hello!</amazon-emotion>
                    {this.options.dummyProps} How are you?
                </p>
            </speak>
        )
    }
    
    reprompt() {
        return (
            <speak>
                <p>How are you?</p>
            </speak>
        )
    }
    
}