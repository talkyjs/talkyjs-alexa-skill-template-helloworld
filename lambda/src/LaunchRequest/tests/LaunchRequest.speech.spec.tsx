import { HandlerInputCreator } from '@ask-utils/test';
import { LaunchRequestScript } from '../LaunchRequest.speech'

describe('LaunchRequestScript', () => {
  
  it('should return false when given a not IntentRequest', async () => {
    const handlerInput = new HandlerInputCreator().createIntentRequest({
      name: "LaunchRequest",
      confirmationStatus: 'NONE'
    });
    const script = new LaunchRequestScript(handlerInput, {
      dummyProps: "It's a nice development."
  });
    expect(script.createResponse()).toMatchSnapshot();
  });
  
});
