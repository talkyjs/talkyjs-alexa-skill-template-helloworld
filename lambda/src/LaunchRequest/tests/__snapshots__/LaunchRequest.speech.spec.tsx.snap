// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchRequestScript should return false when given a not IntentRequest 1`] = `
Object {
  "outputSpeech": Object {
    "ssml": "<speak><p><amazon:emotion name=\\"excited\\" intensity=\\"high\\">Hello!</amazon:emotion>It's a nice development. How are you?</p></speak>",
    "type": "SSML",
  },
  "reprompt": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p>How are you?</p></speak>",
      "type": "SSML",
    },
  },
  "shouldEndSession": false,
}
`;
